tests/migrations/test_operations.py

```diff
@@ -1288,6 +1297,120 @@ def assertIdTypeEqualsFkType():
             operation.database_backwards("test_alflpkfk", editor, new_state, project_state)
         assertIdTypeEqualsFkType()
 
+    def test_alter_field_fk_to_o2o(self):
+        project_state = self.set_up_test_model('test_alflfko2o', related_model=True)
+        self.apply_operations(
+            'test_alflfko2o',
+            project_state,
+            operations=[migrations.AlterField('Rider', 'pony', models.OneToOneField('Pony', models.CASCADE))],
+        )
+
+        Pony = project_state.apps.get_model('test_alflfko2o', 'Pony')
+        Rider = project_state.apps.get_model('test_alflfko2o', 'Rider')
+
+        column = Rider._meta.get_field('pony').column
+
+        if connection.features.supports_foreign_keys:
+            self.assertFKExists(Rider._meta.db_table, [column], (Pony._meta.db_table, Pony._meta.pk.column))
+
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Rider._meta.db_table)
+
+        # Assert index replaced with a unique constraint.
```

Comment:
`# The index on ForeignKey is replace with a unique constraint for OneToOneField.`